external/kafka-0-10-token-provider/src/main/scala/org/apache/spark/kafka010/KafkaTokenUtil.scala

```diff
@@ -288,4 +288,18 @@ private[spark] object KafkaTokenUtil extends Logging {
 
     params
   }
+
+  def isConnectorUsingCurrentToken(params: ju.Map[String, Object]): Boolean = {
+    if (params.containsKey(SaslConfigs.SASL_JAAS_CONFIG)) {
+      logDebug("Delegation token used by connector, checking if uses the latest token.")
+      val consumerJaasParams = params.get(SaslConfigs.SASL_JAAS_CONFIG).asInstanceOf[String]
+      val clusterConfig = findMatchingTokenClusterConfig(SparkEnv.get.conf,
```

Comment:
My previous comment is marked "outdated" because you changed this line, but it still applies, so I wanted to make sure it doesn't go unaddressed.

My best suggestion at the moment is to have the cached consumer to do something like:

```
private val tokenConfig = findMatchingTokenClusterConfig(...)
```

When creating a consumer:

```
val updatedConfigs = KafkaConfigUpdater(...).setAuthInfo(tokenConfig)...
```

And also use that `tokenConfig` when calling this method.

Reply:
At the moment just playing with it to see what are the options which would make it better (that's the reason why not resolved the conversation)...

Reply:
Tried out couple of things and ended up in similar solution what you've suggested. I've considered the actual code and couple of further improvements and `clusterConfig` is the key what all the places needs.